### ISSUE TITLE
What's the training clicking strategy?

### ISSUE THREAD

**Issue by sdfghjkyuio** [saic-vul/fbrs_interactive_segmentation#18]
In paper _Interactive image segmentation via backpropagating reﬁnement scheme_, the  user-annotations are  imitated through a clustering strategy when they trained on the sbd dataset. I'm wondering if you have applied the same way to generate clicks when training?

**Comment by sdfghjkyuio** [comment]
“It ﬁnds the dominant type of prediction errors (false positives or false negatives) and generates the next negative or positive click respectively at the point farthest from the boundaries of the corresponding error region. ” Is this strategy used in training or testing?

**Comment by ksofiyuk** [comment]
We simplify points sampling strategy in our experiments. We observed that if points are just sampled randomly, there is no difference in performance compared to more complicated sampling with points clustering. You can check our training sampling strategy in [this](https://github.com/saic-vul/fbrs_interactive_segmentation/blob/c3f579016f7bd1e0f358cb680a8365ccb6f22b6b/isegm/data/points_sampler.py#L64) class.

**Comment by sdfghjkyuio** [comment]
I guess it's in the points_sampler.py🤔

**Comment by sdfghjkyuio** [comment]
Omg we just commented at the same time! Thank you🤩

**Comment by ksofiyuk** [comment]
Yeah, it's a funny coincidence :smiley:. Thank you for your interest in our work.

**Comment by sdfghjkyuio** [comment]
I just met another problem😫: After I trained for one epoch on my own dataset, I checked the instance segmentation images in the experiment folder and found that some positive points (in green) appeared in the background area, which led to an error in predicting. How can I correct it? Thank you soooo much!

**Comment by sdfghjkyuio** [comment]
Should I create a pkl file based on my own dataset, similar to sbd_samples_weights.pkl?

**Comment by ptrvilya** [comment]
Could you please share a sample from visualization folder with erroneous point placement?

**Comment by sdfghjkyuio** [comment]
![Uploading 206949690927739027.png…]()

Like here, the blue points (background points) appear on the person's face (which is the foreground). So if I hope the network to learn to segment just one specific type from the image, I'm not sure if it can learn the characteristics of this type so well due to the wrong clicks during the training.

**Comment by sdfghjkyuio** [comment]
![206949690927739027](https://user-images.githubusercontent.com/67036136/87434248-cd104180-c61c-11ea-9af7-6a6741e93f80.png)
Like here, the blue points (background points) appear on the person's face (which is the foreground). So if I hope the network to learn to segment just one specific type from the image, I'm not sure if it can learn the characteristics of this type so well due to the wrong clicks during the training.

**Comment by ptrvilya** [comment]
It seems to me that this points lie near the border of the ground-truth mask (on the background part of it). I guess these points appear erroneous due to errors in mask.

**Comment by ksofiyuk** [comment]
I combined your mask with the image and Ilia seems to be right. There is no any mistakes.
![87434248-cd104180-c61c-11ea-9af7-6a6741e93f80](https://user-images.githubusercontent.com/5459767/87440729-f8823480-c5fa-11ea-9c84-a28bc6df5bec.jpg)

**Comment by sdfghjkyuio** [comment]
Yes, the mask can match with the image based on the green and blue points. But I am confused whether the blue points are at the right location. I mean it seems that not all the blue points are sampled from the background pixels.

**Comment by ksofiyuk** [comment]
There are no mistakes with points sampling.
If you want train a good interactive segmentation model, you need to train it on a dataset with nearly perfect segmentation masks. It seems that your dataset has some very coarse masks.

**Comment by sdfghjkyuio** [comment]
Actually this image is one example from the sbd dataset...Thank you for your advice, I will check the masks with wrong sampled points.

**Comment by ksofiyuk** [comment]
Yeah, we know, SBD is not good dataset. We use it only for academic research to compare with previous works.

**Comment by sdfghjkyuio** [comment]
Thank you so much for answering my questions!
